### PR TITLE
WT-10377 Add table level tiered shared configuration option

### PR DESCRIPTION
**Pull request by kommiharibabu**
This new table level tiered shared configuration option can be used to override the default connection tiered shared configuration.

The added new table option is not required to be present in the metadata, so this option is stipped from the configuration string while adding the namespace to the metadata.

### CODE REVIEW

File: src/schema/schema_create.c [hunk 1]
```diff
@@ -278,7 +280,17 @@ __create_file(WT_SESSION_IMPL *session, const char *uri, bool exclusive, const c
...
-        WT_ERR(__wt_metadata_insert(session, uri, fileconf));
+
+        /*
+         * Strip the tiered storage shared configuration from the metadata as it is not required to
+         * be persisted.
+         */
+        filecfg[1] = fileconf;
+        filecfg[2] = NULL;
+        strip = "tiered_storage=(shared=),";
+        WT_ERR(__wt_config_merge(session, filecfg, strip, &filestripped));
```
Comment (ddanderson) [inline src/schema/schema_create.c:291]: These lines occur pretty similarly at least 3 or 4 times.  Maybe could be refactored into a function?

**Review by ddanderson — APPROVED** [review]
I think this makes sense, but I defer to @sueloverso on the nuances of metadata management.

Added a suggestion about refactoring.

**Review by sueloverso — COMMENTED** [review]
I've taken a first look at this but really want to look and think more carefully. Stripping out the metadata item is unusual. I want to think if there's a different way to accomplish it as it feels like it could be fragile if it gets forgotten on some code path and leaks into the metadata.

File: src/schema/schema_create.c (hunk 1 above)
Comment (kommiharibabu) [reply src/schema/schema_create.c]: Thanks. I tried to reduce the duplicate code wherever possible.

**Comment by kommiharibabu** [comment]
@sueloverso @ddanderson Yes I agree that stripping out the metadata from the config before it is inserted into the metadata is a complex change. I checked the flows where the `tiered_config` is used in the metadata and stripped in all the flows other than `lsm`. I am happy to switch if we find any other better alternatives to achieve it.

**Review by agorrod — COMMENTED** [review]
I'm supportive of the idea of this change, but I am also respectful that it's a different way of managing persistent table metadata.
I'm not sure what the right means is to have a deliberate conversation about the shift to (sometimes) not including default values in persistent metadata?

File: test/suite/test_union01.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import os, wiredtiger, wttest
+from helper_tiered import TieredConfigMixin, gen_tiered_storage_sources, get_shared_conn_config
+from wtscenario import make_scenarios
+
+StorageSource = wiredtiger.StorageSource  # easy access to constants
+
+# test_union01.py
```
Comment (sueloverso) [inline test/suite/test_union01.py:35]: I'd rather these go into the `test_tiered##.py` category so that it is easy to run all tiered tests.

**Comment by sueloverso** [comment]
There are other places we strip out metadata so I'm happy with the structure of this.

File: src/config/config_collapse.c [hunk 3]
```diff
@@ -307,6 +307,20 @@ __config_merge_cmp(const void *a, const void *b)
     return (ae->gen > be->gen ? 1 : -1);
 }
 
+/*
+ * __wt_config_tiered_shared_strip --
+ *     This function strips the tiered config shared configuration option writing into the metadata
+ *     from the configuration string to avoid data format changes.
+ */
+int
+__wt_config_tiered_shared_strip(WT_SESSION_IMPL *session, const char **cfg, const char **config_ret)
```
[the remark this answers is not in the document]
Comment (kommiharibabu) [reply src/config/config_collapse.c]: Thanks, fixed.

File: src/schema/schema_create.c [hunk 4]
```diff
@@ -278,7 +280,15 @@ __create_file(WT_SESSION_IMPL *session, const char *uri, bool exclusive, const c
             /* Try to recreate the associated metadata from the imported data source. */
             WT_ERR(__wt_import_repair(session, uri, &fileconf));
         }
-        WT_ERR(__wt_metadata_insert(session, uri, fileconf));
+
+        /*
+         * Strip the tiered storage shared configuration from the metadata as it is not required to
```
[the remark this answers is not in the document]
Comment (kommiharibabu) [reply src/schema/schema_create.c]: Fixed as suggested.

File: test/suite/test_union01.py (hunk 2 above)
Comment (kommiharibabu) [reply test/suite/test_union01.py]: Thanks. I changed the test name to follow the existing tiered tests.

**Pull request merged**